src/artifact_providers/github.ts

```diff
@@ -94,13 +94,16 @@ export class GithubArtifactProvider extends BaseArtifactProvider {
       })
     ).data as unknown) as ArtifactList;
 
-    const allArtifacts = artifactResponse.artifacts;
+    const { artifacts } = artifactResponse;
 
     // We need to find the most recent archive where name matches the revision.
-    const foundArtifact = allArtifacts.reduce((result, artifact) =>
-      artifact.name === revision && result.created_at < artifact.created_at
-        ? artifact
-        : result
+    const foundArtifact = artifacts.reduce(
+      (result, artifact) =>
+        artifact.name === revision &&
+        (!result || result.created_at < artifact.created_at)
+          ? artifact
+          : result,
+      null as ArtifactItem | null
```

Comment:
Why use reduce here instead of a for loop? A for loop would be easier to reason about imo.

Reply:
https://kentcdodds.com/blog/array-reduce-vs-chaining-vs-for-loop 🤔

Reply:
(tl;dr Kent prefers chaining. 😅)

Reply:
Eh, I'll go with a loop, sure :)

Reply:
Also, just below in a comment we say that we assume that artifacts are listed in date descending order on this endpoint. If we can assume that then we don't need to compare dates, do we? We can just take the first one we find? Something like this?

```javascript
    for (let artifact of artifacts) {
      if (artifact.name === revision) {
        return artifact
      }   
    );
```